pipeline-scripts/release.groovy

```diff
@@ -460,4 +460,229 @@ def signArtifacts(Map signingParams) {
+                            labelArgs = "-l 'do-not-merge/hold'"
+                            pr_messages << "Please merge within 48 hours of ${errata_url} shipping live OR a Cincinnati-first release."
+
+                            if (prURLs.containsKey('prerelease')) {
+                                pr_messages << "This should provide adequate soak time for prerelease channel PR ${prURLs.prerelease}"
+                            }
+                            if (prURLs.containsKey('fast')) {
+                                pr_messages << "This should provide adequate soak time for fast channel PR ${prURLs.fast}"
+                            }
+
+                            break
+                        default:
+                            error("Unknown prefix: ${prefix}")
+                    }
+
+                    if ( addToUpgradeChannel ) {
+                        pr_messages << "This PR will also enable upgrades from ${releaseName} to releases in ${upgradeChannel}"
+                    }
+
+                    if ( !addToChannel && !addToUpgradeChannel ) {
+                        def pr_info = "No Cincinnati PRs opened for ${prefix}. Might have been done by previous architecture's release build.\n"
+                        echo pr_info
+                        currentBuild.description += pr_info
+                        continue
+                    }
+
+                    withCredentials([string(credentialsId: 'openshift-bot-token', variable: 'access_token')]) {
+                        def messageArgs = ''
+                        for ( String msg : pr_messages ) {
+                            messageArgs += "--message '${msg}' "
+                        }
+
+                        sh """
+                            set -euo pipefail
+                            set -o xtrace
+                            if git checkout origin/${branchName} ; then
+                                echo "The branch ${branchName} already exists in cincinnati-graph-data. No additional PRs will be created."
+                                exit 0
+                            fi
+                            git branch -f ${branchName} origin/master
+                            git checkout ${branchName}
+                            if [[ "${addToChannel}" == "true" ]]; then
+                                echo >> ${channelFile}    # add newline
+                                echo '# Errata: ${errata_url}' >> ${channelFile}    # add link to errata for reference
+                                echo '- ${releaseName}' >> ${channelFile}   # add the entry
+                                git add ${channelFile}
+                            fi
+                            if [[ "${addToUpgradeChannel}" == "true" ]]; then
+                                echo >> ${upgradeChannelFile}    # add newline
+                                echo '# Errata: ${errata_url}' >> ${upgradeChannelFile}    # add link to errata for reference
```

Comment:
same here. although i think this whole comment might be superfluous for an "upgrade from"?

also, this is still going at the end of the file... is OTA team going to be OK with upgrades being intermingled with channel contents?

Reply:
sed'ing it in would be tricky (but I'll admit to not having sed-fu). Given the following file:
```
....
versions:
- 4.2.0
- 4.2.1

# Something really important about 4.3.0
- 4.3.0
```

If I'm inserting 4.2.99 above the first 4.3 instance, my simple approach would yield:
```
versions:
- 4.2.0
- 4.2.1

# Something really important about 4.3.0
- 4.2.99
- 4.3.0
```

Whitespace and comments will wreak havoc unless I make it fairly complex. 

I've added a new commit which tries to keep things separate, but also simple, by adding 4.x-1 versions directly under `versions:` Example PR: https://github.com/jupierce/cincinnati-graph-data/pull/42